src/v/cluster/service.cc

```diff
@@ -93,16 +93,20 @@ service::join_node(join_node_request&& req, rpc::streaming_context&) {
       || (req.latest_logical_version != cluster::invalid_version && req.latest_logical_version < expect_version)) {
         // Our active version is outside the range of versions the
         // joining node is compatible with.
+        bool permit_join = config::node().upgrade_override_checks();
         vlog(
           clusterlog.warn,
-          "Rejecting join request from incompatible node {}, our version {} vs "
+          "join request from incompatible node {}, our version {} vs "
           "their {}-{}",
+          permit_join ? "" : "Rejecting ",
           req.node,
           expect_version,
           req.earliest_logical_version,
           req.latest_logical_version);
```

Comment:
looks like one too many parameters to the format string?

Reply:
Good spot, fixed.